File: src/python/test/readwrite_plasma.py
import pyarrow.plasma as plasma


class ReadWritePlasma:
    def __init__(self, store_path, obj_id):
        self.store_path = store_path
        self.obj_id = obj_id

    def read(self) -> bytes:
        client = _connect(self.store_path)
        obj = client.get(self.obj_id)
        return obj

    def write(self, data: bytes):
        client = _connect(self.store_path)
        client.put(data, self.obj_id)
        return


# Write to the Plasma Store without having a predefined object ID
# Stores the object and returns it's (new) ID
def write_plasma_without_id(store_path: str, data: object) -> bytes:
    client = _connect(store_path)
    obj_id = client.put(data)
    return obj_id


# Helper function for code reuse
# Returns the client
def _connect(store_path):
    client = plasma.connect(store_path, "", False)
    return client

File: src/java/test/ReadWritePlasma.kt
package test

import PlasmaStore

class ReadWritePlasma(
    private val store: PlasmaStore,
    private val objectId: ByteArray
): ReadWrite {

    override val format: String = "plasma"

    private val client = store.createClient()

    override fun read(): ByteArray {
        return client.get(objectId, 1000, false)
    }

    override fun write(data: ByteArray) {
        if (client.contains(objectId)) {
            client.delete(objectId)
            client.release(objectId)
        }

        client.put(objectId, data, null)
    }
}
